### PR TITLE
fix flixhq show searching

### PR DESCRIPTION
**Pull request by erynith**
- [x] I have read and agreed to the [code of conduct](https://github.com/movie-web/movie-web/blob/dev/.github/CODE_OF_CONDUCT.md).
 - [x] I have read and complied with the [contributing guidelines](https://github.com/movie-web/movie-web/blob/dev/.github/CONTRIBUTING.md).
 - [ ] What I'm implementing was assigned to me and is an [approved issue](https://github.com/movie-web/movie-web/issues?q=is%3Aopen+is%3Aissue+label%3Aapproved). For reference, please take a look at our [GitHub projects](https://github.com/movie-web/movie-web/projects).
 - [x] I have tested all of my changes.

### CODE REVIEW

File: src/providers/sources/flixhq/search.ts [hunk 1]
```diff
@@ -18,16 +18,18 @@ export async function getFlixhqId(ctx: ScrapeContext, media: MovieMedia | ShowMe
...
       return {
         id,
         title,
         year: parseInt(year, 10),
+        seasons,
       };
     });
 
-  const matchingItem = items.find((v) => v && compareMedia(media, v.title, v.year));
+  const matchingItem = items.find((v) => v && (media.type === 'movie' ? compareMedia(media, v.title, v.year) : compareTitle(media.title, v.title) && media.season.number === v.seasons || media.season.number < v.seasons || media.season.number < v.seasons + 1));
```
Comment (mrjvs) [inline src/providers/sources/flixhq/search.ts:32]: You can simplify the check `a === b || a < b || a < b + 1` to `a < b + 1`.

Also I think if would be better write out the condition written in .find() into multiple lines. One big ternary statement is a bit messy to read.

**Review by mrjvs — CHANGES_REQUESTED** [review]
Seems to work nicely, just two code formatting comments and should be ready to be merged

File: src/providers/sources/flixhq/search.ts (hunk 1 above)
Comment (erynith) [reply src/providers/sources/flixhq/search.ts]: ez

Comment (mrjvs) [reply src/providers/sources/flixhq/search.ts]: > Also I think if would be better write out the condition written in .find() into multiple lines. One big ternary statement is a bit messy to read.

you forgot this part

Comment (erynith) [reply src/providers/sources/flixhq/search.ts]: how do i do that 🙏

Comment (mrjvs) [reply src/providers/sources/flixhq/search.ts]: you turn stuff like `lst.find(v=>v&&v===c?v.a===5:v.b===4` into stuff like 

```ts
lst.find(v => {
  if (!v) return false;
  
  if (v === c) {
    if (v.a !== 5) return false;
    return true;
  }
  
  if (v.b !== 4) return false;
  return true;
});
```

Comment (erynith) [reply src/providers/sources/flixhq/search.ts]: never knew that, pretty cool

**Review by mrjvs — CHANGES_REQUESTED** [review]
Linting has failed, I recommend running autoformat and setting up your editor to yell at you for linting errors.

**Comment by erynith** [comment]
> Linting has failed, I recommend running autoformat and setting up your editor to yell at you for linting errors.

i think i fixed it tbh idk

**Review by mrjvs — APPROVED** [review]

**Pull request merged**